agent/resources/resources_linux.go

```diff
@@ -16,27 +16,41 @@
 package resources
 
 import (
+	"os"
+	"path/filepath"
+	"strconv"
+
 	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/config"
 	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
 	"github.com/cihub/seelog"
 	"github.com/containerd/cgroups"
 	"github.com/pkg/errors"
 )
 
+const (
+	memorySubsystem    = "/memory"
+	memoryUseHierarchy = "memory.use_hierarchy"
+	rootReadOnly       = 400
+)
+
 // cgroupWrapper implements the Resource interface
 type cgroupWrapper struct {
-	control cgroup.Control
+	control    cgroup.Control
+	ioutil     ioutilwrapper.IOUtil
```

Comment:
Why should `ioutil` be part of the `cgroupWrapper` object? How about creating the object, when its actually used? (only once?)